packages/nx/src/command-line/migrate/migrate-ui-api.ts

```diff
@@ -171,6 +171,16 @@ export async function runSingleMigration(
         cwd: workspacePath,
         encoding: 'utf-8',
       });
+      // The revision changes after the amend, so we need to update it
```

Comment:
Previously, the recorded `ref` in `migrations.json` were all wrong since they do not include the amended changes to `migrations.json`. This fixes an existing issue.